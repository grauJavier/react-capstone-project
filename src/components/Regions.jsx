import RegionsCards from './RegionsCards';
import CHILE from '../redux/locationData';

const Regions = () => {
  const locationData = CHILE.country;

  return (
    <>
      <nav className='nav'>
        <h1 className='nav__title'>Air Pollution</h1>
      </nav>
      <header className='header'>
        <img src={locationData.img} className='header__img' />
        <div>
          <h3 className='header__name'>{locationData.name}</h3>
          <h4 className='header__stats'></h4>
        </div>
      </header>
      <main className='main'>
        <div className='main__subtitle-container'>
          <h2 className='main__subtitle'>Stats by Region</h2>
        </div>
        <section className='main__section'>
          <RegionsCards />
        </section>
      </main>
    </>
  );  
};

export default Regions;
