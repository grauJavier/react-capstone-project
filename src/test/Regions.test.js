import React from 'react';
import { render, screen } from '@testing-library/react';
import { Provider } from 'react-redux';
import configureMockStore from 'redux-mock-store';
import Regions from '../components/Regions';

jest.mock('../redux/locationData', () => ({
  country: {
    img: 'path/to/image',
    name: 'Chile',
  },
}));

const mockStore = configureMockStore();
const initialState = {}; // Set your initial state if needed
const store = mockStore(initialState);

describe('Regions Component', () => {
  test('renders Regions component with correct elements', () => {
    render(
      <Provider store={store}>
        <Regions />
      </Provider>,
    );

    // Your existing test assertions
    // Check if the component with className "nav__title" exists
    const navTitleElement = screen.getByText(/Air/i);
    const h2Element = screen.getByText(/Stats/i);
    expect(navTitleElement).toBeInTheDocument();
    expect(h2Element).toBeInTheDocument();
    // Check if the h2 element with text "Stats by Region" exists
    const subtitleElement = screen.getByText(/Stats by Region/i);
    expect(subtitleElement).toBeInTheDocument();

    // Take a snapshot of the rendered component
    expect(screen.container).toMatchSnapshot();
  });

  test('renders H2 elements with correct text', () => {
    render(
      <Provider store={store}>
        <Regions />
      </Provider>,
    );
    // Your existing test assertions
    // Check if the component with className "nav__title" exists
    const h2Element = screen.getByText(/Stats/i);
    expect(h2Element).toBeInTheDocument();
    // Check if the h2 element with text "Stats by Region" exists
    const subtitleElement = screen.getByText(/Stats by Region/i);
    expect(subtitleElement).toBeInTheDocument();

    // Take a snapshot of the rendered component
    expect(screen.container).toMatchSnapshot();
  });
});
