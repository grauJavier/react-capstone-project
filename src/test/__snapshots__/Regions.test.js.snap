// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Regions Component renders H2 elements with correct text 1`] = `undefined`;

exports[`Regions Component renders Regions component with correct elements 1`] = `undefined`;
